### ISSUE TITLE
Annotated Enums inside a dataclass using a match can falsely report reportMatchNotExhaustive

### ISSUE THREAD

**Issue by gitpushdashf** [microsoft/pyright#4977]
**Describe the bug**

Annotated Enums inside a dataclass using a match can falsely report reportMatchNotExhaustive.

**Screenshots or Code**

```python
from dataclasses import dataclass
from enum import Enum, auto
from typing import Annotated


class MyEnum(Enum):
    FIRST = auto()
    SECOND = auto()


@dataclass
class ThingWithMyEnum:
    my_enum: MyEnum
    my_enum_annotated: Annotated[MyEnum, str]


def matcharoo(something: ThingWithMyEnum) -> None:
    match something.my_enum:
        case MyEnum.FIRST:
            print("It's first")
        case MyEnum.SECOND:
            print("It's second")


def matcharoo_annotated(something: ThingWithMyEnum) -> None:
    match something.my_enum_annotated:
        case MyEnum.FIRST:
            print("It's first")
        case MyEnum.SECOND:
            print("It's second")
```

```shell
pyright 1.1.304
pyrightbug.py
  pyrightbug.py:26:11 - error: Cases within match statement do not exhaustively handle all values
    Unhandled type: "MyEnum"
    If exhaustive handling is not intended, add "case _: pass" (reportMatchNotExhaustive)
1 error, 0 warnings, 0 informations
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by gitpushdashf** [comment]
> Thanks for the bug report. This will be fixed in the next release.

Awesome! Thank you so much!!

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.305, which I just published. It will also be included in a future release of pylance.
